### ISSUE TITLE
Markdown in "[[test]]: test" pattern won't be tokenized

### ISSUE THREAD

**Issue by dragonman225** [nornagon/remarkable-wikilink#1]
Thank you for creating this library! I'm also trying to build a networked document editor with remarkable, and supporting wikilink is the first key step. I tried this library and it worked in most cases, except for the one I mentioned in the title.

Out of curiosity, I tested several similar patterns to see what is exactly causing problems, here's the result: (also on [codesandbox](https://codesandbox.io/s/remarkable-wikilink-test-tdl7in?file=/src/index.js))

```
The broken case:                  "[[test]]: test"      → produce no tokens, output is empty
Remove the colon:                 "[[test]] test"       → tokenized correctly
Add the second word after:        "[[test]]: test test" → tokenized correctly
Add a non-space character before: "t[[test]]: test"     → tokenized correctly
Add a space before:               " [[test]]: test"     → produce no tokens, output is empty
```

I thought it has something to do with the colon—maybe another parsing rule has higher priority and override this?—so I tried moving the wikilink rule before all other rules, but it still didn't work.

Do you have any ideas about why it is not working? Or any pointers on where I can look into?

**Comment by nornagon** [comment]
ah, that sounds weird! i cargo-culted the tokenization code here from another remarkable plugin, so I don't actually understand it well enough to say for sure what might be going on.

**Comment by dragonman225** [comment]
It turned out that in the broken cases Remarkable called `wikilinkRule` function with `pos` being at the second `[`, so the rule never matched. I created [a quick patch](https://github.com/nornagon/remarkable-wikilink/compare/master...dragonman225:remarkable-wikilink:master), which tries to look one character before `pos` to cover such cases. It is not perfect, though, for example, `[[[test]]` is tokenized to

```json5
[
  {
    "type": "text",
    "content": "[[",
    "level": 0
  },
  {
    "type": "wikilink_open",
    "href": "test",
    "level": 0
  },
  {
    "type": "text",
    "content": "est", // Where is the first "t"?
    "level": 1
  },
  {
    "type": "wikilink_close",
    "level": 0
  }
]
```

---

And I found that I needed the rule to prevent nested content and support [link label](https://en.wikipedia.org/wiki/Help:Link#Piped_link)—may introduce big changes to the code—so I decided to do a full rewrite instead. I'll drop a line here when it's done.
